src/natcap/invest/habitat_quality.py

```diff
@@ -817,43 +1064,45 @@ def raster_pixel_count(raster_path):
     return counts
 
 
-def map_raster_to_dict_values(
-        key_raster_path, out_path, attr_dict, field, out_nodata, values_required):
-    """Creates a new raster from 'key_raster' where the pixel values from
-       'key_raster' are the keys to a dictionary 'attr_dict'. The values
-       corresponding to those keys is what is written to the new raster. If a
-       value from 'key_raster' does not appear as a key in 'attr_dict' then
-       raise an Exception if 'raise_error' is True, otherwise return a
-       'out_nodata'
-
-       key_raster_path - a GDAL raster path dataset whose pixel values relate to
-                     the keys in 'attr_dict'
-       out_path - a string for the output path of the created raster
-       attr_dict - a dictionary representing a table of values we are interested
-                   in making into a raster
-       field - a string of which field in the table or key in the dictionary
-               to use as the new raster pixel values
-       out_nodata - a floating point value that is the nodata value.
-       raise_error - a string that decides how to handle the case where the
-           value from 'key_raster' is not found in 'attr_dict'. If 'raise_error'
-           is 'values_required', raise Exception, if 'none', return 'out_nodata'
-
-       returns - a GDAL raster, or raises an Exception and fail if:
-           1) raise_error is True and
-           2) the value from 'key_raster' is not a key in 'attr_dict'
-    """
+def _map_raster_to_dict_values(
+        key_raster_path, out_path, attr_dict, field, out_nodata,
+        values_required):
```

Comment:
I commented above too, but if I understand what's happening here I think it might be more straightforward to use `reclassify_raster` directly in your code above and then you can skip this whole function.